### ISSUE TITLE
Delete code surrounded by opt_panama and Panama flags

### ISSUE THREAD

**Issue by babsingh** [eclipse-openj9/openj9#11269]
`opt_panama`: `J9VM_OPT_PANAMA` surrounds the native code
`Panama`: JCL flag surrounds the Java code

The above flags highlight OpenJ9's implementation for the `NativeMethodHandle` and `Foreign Linker API` (JEP389). 

The above OpenJ9 implementation can be deleted since it will be superseded by the OpenJDK implementation.

The `MHInterpreter` code can be retained since it can be repurposed.

**Comment by babsingh** [comment]
fyi @tajila

**Comment by babsingh** [comment]
https://github.com/eclipse/openj9/pull/11267#issuecomment-733239213:

I tried to turn-on the `opt_panama` and `Panama` (JCL) flags in order to enable the OpenJ9 `NativeMethodHandle` support. But, I ended up facing many issues:
1. `jdk.internal.nicl` and `java.nicl` packages no longer exist in JDK16. Finding their new equivalent representation will take some time. 
2. `MethodHandles` class no longer has the `findNative*` methods.
3. `Library` class is no longer available; potentially renamed to `NativeLibrary` and relocated.
4. `gcc` error: `Jit.hpp` in `JavaMethodCallbackGen.cpp` cannot be found.
5. ... the list goes on
